### PR TITLE
HTTP/3: Test abort at end of request

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/dotnet/runtime/issues/72619

### CODE REVIEW

File: src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs [hunk 1]
```diff
@@ -537,6 +537,58 @@ public async Task POST_ServerAbort_ClientReceivesAbort(HttpProtocols protocol)
...
+            Logger.LogInformation("Server aborting.");
+            context.Abort();
+        }, protocol: HttpProtocols.Http3);
+
+        using (var host = builder.Build())
+        using (var client = HttpHelpers.CreateClient())
+        {
+            await host.StartAsync().DefaultTimeout();
+
+            for (var i = 0; i < 1000; i++)
```
Comment (halter73) [inline src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs:561]: How long does this test take? I don't think tests that run on every CI run and every developer's machines are the right place to do stress testing.

Even if this only takes a second, the road to a long-running test suite are tests like these that are needlessly repetitive. I brought all of Kestrel's 1000s of tests down from many minutes to under 30 seconds when I first moved most of them in memory. That's also why most of our timeout tests use a fake clock and ticks.

We should get this running on dedicated stress infrastructure.

Comment (JamesNK) [reply src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs:561]: It takes 1.3 seconds.

I lowered requests to 100, and it takes 0.16s.

File: src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs [hunk 2]
```diff
@@ -537,6 +537,58 @@ public async Task POST_ServerAbort_ClientReceivesAbort(HttpProtocols protocol)
...
+    {
+        // Arrange
+        var builder = CreateHostBuilder(async context =>
+        {
+            Logger.LogInformation("Server writing content.");
+            await context.Response.Body.WriteAsync(new byte[16]);
+
+            // Note that there is a race here on what is sent before the abort is processed.
+            // Abort may happen before or after response headers have been sent.
+            Logger.LogInformation("Server aborting.");
```
Comment (Tratcher) [inline src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs:552]: Do you want to wait until the response is received before aborting? Or are intentionally covering both cases here?

**Review by Tratcher — APPROVED** [review]

File: src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs (hunk 2 above)
Comment (JamesNK) [reply src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs:552]: It's intentional. There are other tests that wait for the response.

**Pull request merged**